optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/bavet/uni/BavetFromUniTuple.java

```diff
@@ -16,12 +16,19 @@
 
 package org.optaplanner.core.impl.score.stream.bavet.uni;
 
+import java.util.ArrayList;
+import java.util.List;
+
+import org.optaplanner.core.impl.score.stream.bavet.common.BavetAbstractTuple;
+
 public final class BavetFromUniTuple<A> extends BavetAbstractUniTuple<A> {
 
     private final BavetFromUniNode<A> node;
     private final A factA;
+    private final List<BavetAbstractTuple> childTupleList;
 
     public BavetFromUniTuple(BavetFromUniNode<A> node, A factA, int childTupleListSize) {
+        this.childTupleList = new ArrayList<>(childTupleListSize);
```

Comment:
Idea: as we apparrantly know the maximum childTupleListSize for a fromUni tuple,
we probably know it for the filterUni tuples too?

For example:
- from(A)
-- filter(A.b != null)
---  join(C).penalize
--- join(D).penalize
--- filter(E).penalize
-- filter(F).penalize

the from has max 2 child tuples, the filter has max 3. Even if the turns out less that than, the 3 is still a very good ArrayList capacity.

Reply:
You keep assuming that there is node sharing. That is a very poor assumption, IMO, outside of the from nodes.
The bar to actually get a node shared is very high, and people will never meet it.

(Unless they write some complicated and non-straight-forward constraint providers.)